File: _code_examples/kamon-annotation-examples/src/main/scala/kamon/annotation/examples/Annotated.scala
// tag:activation:start
import kamon.annotation.EnableKamon
import kamon.annotation.Count

@EnableKamon
class Annotated {

	@Count(name="counter")
	def countedMethod:Unit = {}
}	
// tag:activation:end


package object el {
// tag:el-support:start
import kamon.annotation.EnableKamon
import kamon.annotation.Count

@EnableKamon
class Annotated {

	@Count(name="counter", tags = "#{'counter':'my-awesome-counter', 'environment':'prod'}")
	def countedMethod:Unit = {}	
}
// tag:el-support:end	
}


package object elName {
// tag:el-support-instrument-name:start
import kamon.annotation.EnableKamon
import kamon.annotation.Count

@EnableKamon
case class Annotated(id: Long) {

 	@Count(name = "${'count:' += this.id}", tags = "#{'my-awesome-counter':'1', 'env':'prod'}")
  	def countedMethod(): Unit = {}
	
}
// tag:el-support-instrument-name:end	
}


package object static {
// tag:static-methods:start
import kamon.annotation.EnableKamon
import kamon.annotation.Count

@EnableKamon
object AnnotatedObject {

 	@Count(name = "counter")
  	def countedMethod(): Unit = {}
	
}	
// tag:static-methods:end
}

File: _code_examples/kamon-annotation-examples/src/main/java/kamon/annotation/examples/el/name/Annotated.java
package kamon.annotation.examples.el.name;

// tag:el-support-instrument-name:start
import kamon.annotation.EnableKamon;
import kamon.annotation.Count;

@EnableKamon
public class Annotated {

	private final Long id;

	public Annotated(Long id) {
		this.id = id;
	}

 	@Count(name = "${'count:' += this.id}", tags = "#{'my-awesome-counter':'1', 'env':'prod'}")
  	public void countedMethod() {}
	
}	
// tag:el-support-instrument-name:end